app/controllers/foods_controller.rb

```diff
@@ -1,22 +1,32 @@
 class FoodsController < ApplicationController
   def index
+    @food = Food.all
+  end
 
+  def show
+    @food = Food.find(params[:id])
+  rescue ActiveRecord::RecordNotFound
+    flash[:alert] = "The food you were looking for could not be found."
+    redirect_to foods_path
   end
+
   def new
-    @food = Food.new(food_params)
+    @food = Food.new
   end
-def create
-  @food = Food.create(food_params)
-  if food.save
-    redirect_to foods_path, notice: "Food create successfully"
-  else
-    flash["alert"] = "Food not created"
-end
+
+  def create
+    @food = Food.new(food_params)
+    @user = current_user
```

Comment:
The food object is already stocked in the @food instance variable. so, it has the user_id column which is nil for now. to assign the current_user as the owner of the food you need to get the food object which is the @food then assign the user to it like :
@food.user =  current_user. know that in programming the right object assigned always his value to the left one when we're using =